java/com/google/location/lbs/activity/audioset/dataprocessing/CsvParser.java

```diff
@@ -0,0 +1,244 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Arrays;
+
+/**
+ * CSV Class
+ *
+ * <p>CSV is a class for the reading of a dataset in a csv format.
+ *
+ * @author Carver Forbes
+ * @version 0.1
+ */
+public class CsvParser {
+
+  private File file;
+  private ArrayList<String> ytidList;
+  private HashMap<String, IncompleteExample.Example> idToProtoMap;
+  private HashMap<String, String> idToUrlMap;
```

Comment:
generally it's a good idea to use interfaces like List, Map in declaration

Reply:
In this case, an efficient data structure would be a map with id being the key to the `IncompleteExample.Example` proto. The url we can always deduce from the ID since the url prefix is constant. We can always get the youtube id from the actual key in the `Map`

Reply:
Ok. That's true. Let me remove that part. Thank you.